src/hotspot/share/gc/shenandoah/shenandoahHeapRegion.cpp

```diff
@@ -98,6 +99,7 @@ void ShenandoahHeapRegion::make_regular_allocation() {
     case _empty_uncommitted:
       do_commit();
     case _empty_committed:
+      set_affiliation(YOUNG_GENERATION);
```

Comment:
It's not clear to me why we would set the affiliation here. Do we know this region isn't going to be used for evacuations in old gen?

Reply:
This is to fix full single-generational collections. But yes, not good enough in the long run. => I need to put a TODO note to remind us of later repairing proper full collections in generational mode.

Reply:
The TODO is in the next commit I filed for this PR. The text for it is taken from a similar place that needs to be fixed in a very similar way.